graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/predicate/SorterHelper.java

```diff
@@ -79,47 +78,33 @@
     }
 
     static {
-        SORT_BY_DIRECTION.put(SortType.ASC, ((source, fieldName, fieldValue, ignoreCase, environment) -> {
+        FieldSorterAlgorithm fieldSorterAlgorithm = (source, fieldName, fieldValue, ignoreCase, environment) -> {
             Object value = environment.getFieldValue(source, fieldName);
             if (value == null && fieldValue == null) {
                 return 0;
-            } else if (value == null && fieldValue != null) {
+            } else if (value == null) {
                 return -1;
-            } else if (value != null && fieldValue == null) {
+            } else if (fieldValue == null) {
                 return 1;
             } else if (fieldValue instanceof Boolean) {
-                return Boolean.compare((Boolean)value, (Boolean)fieldValue);
+                return Boolean.compare((Boolean) value, (Boolean) fieldValue);
             } else if (fieldValue instanceof Long) {
-                return Long.compare((Long)value, (Long)fieldValue);
+                return Long.compare((Long) value, (Long) fieldValue);
             } else if (fieldValue instanceof Double) {
-                return Double.compare((Double)value, (Double)fieldValue);
+                return Double.compare((Double) value, (Double) fieldValue);
             } else if (fieldValue instanceof Integer) {
-                return Integer.compare((Integer)value, (Integer)fieldValue);
+                return Integer.compare((Integer) value, (Integer) fieldValue);
+            } else if (fieldValue instanceof String) {
+                return ignoreCase ? ((String)value).compareToIgnoreCase(((String)fieldValue)) : ((String)value).compareTo(((String)fieldValue));
```

Comment:
How do we handle dates comparison, will String comparison be sufficient for our use cases?

Reply:
yes, dates are returned as string for graphql, so no specific type -  and they are comparable in lexicographical order